TF_Service_dotNet/TouchFree/HandManager.cs

```diff
@@ -58,21 +60,42 @@ public class HandManager : IHandManager
 
         private int handsLastFrame;
 
+        private Leap.Image lastImage;
+
         public HandFrame RawHands { get; private set; }
         public List<Hand> PreConversionRawHands { get; private set; }
         public bool RawHandsUpdated { get; private set; }
 
-        public HandManager(ITrackingConnectionManager _trackingConnectionManager, IConfigManager _configManager, IVirtualScreen _virtualScreen)
+        public HandManager(ITrackingConnectionManager _trackingManager, IConfigManager _configManager, IVirtualScreen _virtualScreen, IUpdateBehaviour _updateBehaviour)
         {
             handsLastFrame = 0;
 
-            ConnectionManager = _trackingConnectionManager;
+            ConnectionManager = _trackingManager;
             virtualScreen = _virtualScreen;
             configManager = _configManager;
-            ConnectionManager.Controller.FrameReady += Update;
-            ConnectionManager.Controller.ImageReady += UpdateRawHands;
-            _configManager.OnPhysicalConfigUpdated += UpdateTrackingTransform;
-            UpdateTrackingTransform(_configManager.PhysicalConfig);
+
+            if (_configManager != null)
+            {
+                _configManager.OnPhysicalConfigUpdated += UpdateTrackingTransform;
+                UpdateTrackingTransform(_configManager.PhysicalConfig);
+            }
+
+            if (ConnectionManager != null)
+            {
+                ConnectionManager.Controller.FrameReady += Update;
+                ConnectionManager.Controller.ImageReady += StoreImage;
+            }
+            _updateBehaviour.OnSlowUpdate += UpdateRawHands;
+        }
+
+        public bool TrackingServiceConnected()
+        {
+            return ConnectionManager?.Controller?.IsServiceConnected ?? false;
+        }
+
+        public bool CameraConnected()
+        {
+            return ConnectionManager?.Controller?.Devices?.ActiveDevice != null;
```

Comment:
Unreferenced (should be private at least?)

Reply:
Removed these as they aren't used